lambdas/raster_tiler/lambda_function.py

```diff
@@ -99,20 +99,28 @@ def apply_annual_loss_filter(
         year <= _end_year - 2000
     )
 
-    red: ndarray = np.ones(intensity.shape).astype("uint8") * 228
+    # Construct RGBA bands with floating point values for precision
+    red: np.ndarray = np.ones(intensity.shape).astype("float32") * 228
     green: ndarray = (
-        np.ones(intensity.shape) * 102
+        np.ones(intensity.shape).astype("float32") * 102
         + (72 - zoom)
         - (scaled_intensity * (3 / max(zoom, 1)))
-    ).astype("uint8")
+    )
     blue: ndarray = (
-        np.ones(intensity.shape) * 153 + (33 - zoom) - (intensity / max(zoom, 1))
-    ).astype("uint8")
+        np.ones(intensity.shape).astype("float32") * 153 
+        + (33 - zoom) 
+        - (intensity / max(zoom, 1))
+    )
     alpha: ndarray = (
         (scaled_intensity if zoom < 13 else intensity) * start_year_mask * end_year_mask
-    ).astype("uint8")
+    ).astype("float32")
 
-    return np.array([red, green, blue, alpha])
+    # Ensure GBA values are in [0, 255] range
+    green = np.clip(green, 0, 255).astype("uint8")
```

Comment:
Do you want to clip red to 255 as well?  If not, can you put in a comment on why you don't need to clip it.  (For instance, maybe intensity.shape is only a value between 0 and 1, so you know red is already between 0 and 228?)